aidial_rag/retrievers/colpali_retriever/colpali_model_resource.py

```
@@ -0,0 +1,281 @@
+    def __init__(
+        self,
+        config: ColpaliModelResourceConfig | None,
+        colpali_index_config: ColpaliIndexConfig | None,
+    ):
+        self.lock = threading.Lock()
+        self.model_resource_config: ColpaliModelResourceConfig | None = None
+        self.colpali_index_config: ColpaliIndexConfig | None = None
+        self.index_config: ColpaliIndexConfig | None = None
+        self.model = None
+        self.device: torch.device | None = None
+        self.processor = None
+        self.batch_processor: Optional[ColpaliBatchProcessor] = None
+        self.query_batch_processor: Optional[ColpaliBatchProcessor] = None
+        if colpali_index_config is not None and config is not None:
+            self.__set_config(config)
+
+    def __set_config(self, config: ColpaliModelResourceConfig):
+        config.validate_model_name()
+
+        with self.lock:
+            if self.model_resource_config == config:
+                return
+            self.model_resource_config = config
+            device = autodetect_device().value
+            self.device = torch.device(device)
+
+            model_class, processor_class = get_model_processor_classes(
+                config.model_name
+            )
+
+            # Check if local model path exists otherwise use Hugging Face
+            model_name = config.model_name
+            cache_path = None
+            print(f"COLPALI_MODELS_BASE_PATH: {COLPALI_MODELS_BASE_PATH}")
+            if COLPALI_MODELS_BASE_PATH:
+                local_model_path = get_model_local_path(
+                    COLPALI_MODELS_BASE_PATH, model_name
+                )
+                if os.path.exists(local_model_path):
+                    model_name = local_model_path
+                    cache_path = get_model_cache_path(local_model_path)
+                    print(f"loading model from local path: {local_model_path}")
+                else:
+                    print("loading model from hugging face")
```

If the local path is specified and the model is expected to be loaded from local path, I think, we should throw an error here instead of a fallback to the huggingface.

---

changed behaviour, now it throws an error